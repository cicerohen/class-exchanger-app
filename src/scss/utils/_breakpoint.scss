@function lower-bound($range) {
  @if length($range) <= 0 {
    @return 0;
  }
  @return nth($range,1);
}

@function upper-bound($range) {
  @if length($range) < 2 {
    @return 999999999999;
  }
  @return nth($range, 2);
}

$small-range:(0em, 39.9375em);
$medium-range:(40em, 63.9375em);
$large-range:(64em, 74.9375em);

$screen: "only screen" !default;
$small-only: "#{$screen} and (max-width: #{upper-bound($small-range)})" !default;
$medium-only: "#{$screen} and (min-width: #{lower-bound($medium-range)}) and (max-width: #{upper-bound($medium-range)})" !default;
$medium-up: "#{$screen} and (min-width: #{lower-bound($medium-range)})" !default;
$large-only: "#{$screen} and (min-width: #{lower-bound($large-range)}) and (max-width: #{upper-bound($large-range)})" !default;
$large-up: "#{$screen} and (min-width: #{lower-bound($large-range)})" !default;
