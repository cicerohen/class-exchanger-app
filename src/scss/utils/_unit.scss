@function strip-unit($num) {
  @return $num / ($num * 0 + 1);
}

@function rem-calc($value, $base: null) {
    
  @if $base == null {
    $base: 62.5%;
  }

  @if unit($base) == '%' {
    $base: ($base / 100%) * 16px;
  }

  @if type-of($value) != 'number' {
    @warn inspect($value) + ' was passed to rem-calc(), which is not a number.';
    @return $value;
  }

  @if unit($value) != 'rem' {
    $value: strip-unit($value) / strip-unit($base) * 1rem;
  }

  @if $value == 0rem {
    $value: 0;
  }
  
  @return $value;

}
