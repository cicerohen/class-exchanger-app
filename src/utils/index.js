import { Iterable, fromJS } from 'immutable';
import { createBrowserHistory } from 'history';

export const asImmutable = obj => (Iterable.isIterable(obj) ? obj : fromJS(obj));
export const emptyMap = asImmutable({});

export const buildActions = (moduleName, prefix, suffix) => ({
    REQUESTED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}REQUESTED`,
    STARTED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}STARTED`,
    SUCCEEDED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}SUCCEEDED`,
    FAILED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}FAILED`,
    FULFILLED: `${moduleName}/${prefix}_${suffix || ''}${suffix ? '_' : ''}FULFILLED`,
});

export const toArrayPath = path => path.split('/').filter(p => p.replace(' ', '').length > 0);

export const getHistory = () => {
    window.browserHistory = window.browserHistory || createBrowserHistory();
    return window.browserHistory;
};

export const isDevEnv = () => {
  const nodeEnv = process.env.NODE_ENV;
  return nodeEnv !== 'production' && nodeEnv !== 'test';
};
