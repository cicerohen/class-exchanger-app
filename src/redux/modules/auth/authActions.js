import { buildActions } from '../../../utils';
import MODULE_NAME from './authConstants';

export const AUTH_STATE_CHANGED = `${MODULE_NAME}/AUTH_STATE_CHANGED`;

export const SIGN_IN_WITH_EMAIL = buildActions(MODULE_NAME, 'SIGN_IN_WITH_EMAIL');
export const SIGN_UP_WITH_EMAIL = buildActions(MODULE_NAME, 'SIGN_UP_WITH_EMAIL');
export const SIGN_OUT = buildActions(MODULE_NAME, 'SIGN_OUT');

export const USER_SAVED = `${MODULE_NAME}/USER_SAVED`;
export const USER_REMOVED = `${MODULE_NAME}/USER_REMOVED`;

export const authStateChange = user => ({
  type: AUTH_STATE_CHANGED,
  payload: {
    user,
  },
});

/* signIn */
export const signInWithEmail = (email, password) => ({
    type: SIGN_IN_WITH_EMAIL.REQUESTED,
    payload: {
        email,
        password,
    },
});

export const signInWithEmailFulfilled = data => ({
    type: SIGN_IN_WITH_EMAIL.SUCCEEDED,
    payload: {
        data,
    },
});

export const signInWithEmailFailed = error => ({
    type: SIGN_IN_WITH_EMAIL.FAILED,
    payload: {
        error,
    },
});

// signUp
export const signUpWithEmail = (email, password) => ({
    type: SIGN_UP_WITH_EMAIL.REQUESTED,
    payload: {
        email,
        password,
    },
});

export const signUpWithEmailFulfilled = data => ({
    type: SIGN_UP_WITH_EMAIL.SUCCEEDED,
    payload: {
        data,
    },
});

export const signUpWithEmailFailed = error => ({
    type: SIGN_UP_WITH_EMAIL.FAILED,
    payload: {
        error,
    },
});

// signOut
export const signOut = () => ({
    type: SIGN_OUT.REQUESTED,
});

export const signOutFulfilled = () => ({
    type: SIGN_OUT.SUCCEEDED,
});

export const signOutFailed = () => ({
    type: SIGN_OUT.FAILED,
});

export const saveUser = ({
    uid, email, displayName, emailVerified,
}) => ({
    type: USER_SAVED,
    payload: {
        uid,
        email,
        displayName,
        emailVerified,
    },
});

export const removeUser = () => ({
    type: USER_REMOVED,
});
