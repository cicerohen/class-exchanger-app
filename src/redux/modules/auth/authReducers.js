import { asImmutable } from '../../../utils';
import * as actions from './authActions';

const initialState = asImmutable({
  user: {},
  isAuthenticated: false,
  isAuthenticating: true,
});

export const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actions.AUTH_STATE_CHANGED: {
      return state.merge({
        isAuthenticated: !!action.payload.user,
        isAuthenticating: false,
      });
    }
    case actions.USER_SAVED: {
      return state.mergeIn(['user'], asImmutable(action.payload));
    }
    case actions.USER_REMOVED: {
      return state.set('user', asImmutable({}));
    }
    default:
      return state;
  }
};
