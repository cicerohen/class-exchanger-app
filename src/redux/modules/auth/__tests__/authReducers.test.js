import * as actions from '../authActions';
import { reducer } from '../authReducers';
import { asImmutable } from '../../../../utils';

const initialState = {
  user: {
    uid: 1,
    email: 'cicero@viana.com',
    displayName: 'Cicero Viana',
    emailVerified: false,
  },
  isAuthenticated: false,
  isAuthenticating: true,
};

describe('authReducers tests', () => {
  it('should store correctly when user is saved', (done) => {
    const nextState = reducer(asImmutable(initialState), actions.saveUser({
      uid: 2,
      email: 'user@email.com',
      displayName: 'User',
      emailVerified: true,
    }));
    expect(nextState.toJS()).toMatchSnapshot();
    done();
  });

  it('should store correctly when user is removed', (done) => {
    const nextState = reducer(asImmutable(initialState), actions.removeUser());
    expect(nextState.toJS()).toMatchSnapshot();
    done();
  });
});