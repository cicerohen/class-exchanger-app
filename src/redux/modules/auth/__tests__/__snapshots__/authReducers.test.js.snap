// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`authReducers tests should store correctly when user is removed 1`] = `
Object {
  "isAuthenticated": false,
  "isAuthenticating": true,
  "user": Object {},
}
`;

exports[`authReducers tests should store correctly when user is saved 1`] = `
Object {
  "isAuthenticated": false,
  "isAuthenticating": true,
  "user": Object {
    "displayName": "User",
    "email": "user@email.com",
    "emailVerified": true,
    "uid": 2,
  },
}
`;
